### PR TITLE
Made a button to make/view reviews in game detail page

### LINKED ISSUE #261: add button to see / make reviews on game detail page
Include a button in the game detail page from #40 that allows a user to make a review of a game. Make a list view for the reviews as well to display a list of reviews.

### PR DESCRIPTION
**Pull request by danielngira**
Related Issue: #261 

Tasks I completed in this PR:

- Made a form ReviewForm for creating new reviews
- Updated the GameDetailView with a FormMixin and LoginRequiredMixin to allow logged-in users to create reviews for games within the game detail page.
- Made a ReviewListView and a game-review template. Also updated the URLs to include a URL to the game-reviews page for a specific game
- Updated the game-detail template page to include a form to create new reviews

This PR depends on PR #274 that will create a new review entity in the models

### CODE REVIEW

**Comment by danielngira** [comment]
https://github.com/uchicago-cs/chigame/assets/81624602/eadcd678-12ef-4159-ac42-0d7bb969694f

You can now leave a review on a game and view a list of a game's public reviews.

File: src/chigame/games/forms.py [hunk 1]
```diff
@@ -25,3 +26,26 @@ class Meta:
         model = Lobby
         fields = ["name", "game", "game_mod_status", "min_players", "max_players", "time_constraint"]
         widgets = {"name": forms.TextInput}
+
+
+class ReviewForm(forms.ModelForm):
```
Comment (rakrusch) [inline src/chigame/games/forms.py:31]: Review form here looks good to me!

File: src/chigame/games/views.py [hunk 2]
```diff
@@ -42,10 +43,34 @@ def get_queryset(self):
         return queryset
 
 
-class GameDetailView(DetailView):
+class GameDetailView(LoginRequiredMixin, FormMixin, DetailView):
```
Comment (rakrusch) [inline src/chigame/games/views.py:46]: Nice job adding reviews onto game detail view!

**Review by rakrusch — APPROVED** [review]
Everything looks good and appears to be working well. Nice job Daniel!

**Comment by danielngira** [comment]
I noticed the template page was not displaying the game's name. In my latest commit, I fixed how it is rendered in the `ReviewListView`

**Review by majorsylvie — CHANGES_REQUESTED** [review]
If not supplying anything to a review, the validation error shows twice: 
image

The titles of reviews are not seen in the list view:
image

Seen more obviously with: 
image

image

Along with that, the numbers are displayed somewhat oddly. 
You use the `/5` instead of `/5.00` even though we are doing 2 decimal place floats from 1-5 for ratings.

Along with that, you still display the rating for reviews without one, leading to this odd feature of the UI:
image

**Please address the following:**
- not displaying `None` ratings
- changing the ratings to show `/5.00` to better reflect the numbers for ratings
- Actually display rating titles if they exist in the ratings-per-game list view

**Issues to create based on enhancing the work in this PR**
- one to add a visual display (such as filled in stars) to reflect the rating 
  - ie: 4.5 == 4 and a half filled in stars
- remove duplicate error messages for review validation
- add extra styling/CSS to the reviews in the review-per-game list view to make them look nicer (whatever that means to you).

Please make the requested changes **and create issues in the backlog for all listed enhancements**

Thank you for your work! this is super cool functionality that you've spearheaded @danielngira. This is quite close to merging!

**Comment by danielngira** [comment]
> If not supplying anything to a review, the validation error shows twice: image
> 
> The titles of reviews are not seen in the list view: image
> 
> Seen more obviously with: image
> 
> image
> Along with that, the numbers are displayed somewhat oddly. You use the `/5` instead of `/5.00` even though we are doing 2 decimal place floats from 1-5 for ratings.
> 
> Along with that, you still display the rating for reviews without one, leading to this odd feature of the UI: image
> 
> **Please address the following:**
> 
> * not displaying `None` ratings
> * changing the ratings to show `/5.00` to better reflect the numbers for ratings
> * Actually display rating titles if they exist in the ratings-per-game list view
> 
> **Issues to create based on enhancing the work in this PR**
> 
> * one to add a visual display (such as filled in stars) to reflect the rating
>   
>   * ie: 4.5 == 4 and a half filled in stars
> * remove duplicate error messages for review validation
> * add extra styling/CSS to the reviews in the review-per-game list view to make them look nicer (whatever that means to you).
> 
> Please make the requested changes **and create issues in the backlog for all listed enhancements**
> 
> Thank you for your work! this is super cool functionality that you've spearheaded @danielngira. This is quite close to merging!



> If not supplying anything to a review, the validation error shows twice: image
> 
> The titles of reviews are not seen in the list view: image
> 
> Seen more obviously with: image
> 
> image
> Along with that, the numbers are displayed somewhat oddly. You use the `/5` instead of `/5.00` even though we are doing 2 decimal place floats from 1-5 for ratings.
> 
> Along with that, you still display the rating for reviews without one, leading to this odd feature of the UI: image
> 
> **Please address the following:**
> 
> * not displaying `None` ratings
> * changing the ratings to show `/5.00` to better reflect the numbers for ratings
> * Actually display rating titles if they exist in the ratings-per-game list view
> 
> **Issues to create based on enhancing the work in this PR**
> 
> * one to add a visual display (such as filled in stars) to reflect the rating
>   
>   * ie: 4.5 == 4 and a half filled in stars
> * remove duplicate error messages for review validation
> * add extra styling/CSS to the reviews in the review-per-game list view to make them look nicer (whatever that means to you).
> 
> Please make the requested changes **and create issues in the backlog for all listed enhancements**
> 
> Thank you for your work! this is super cool functionality that you've spearheaded @danielngira. This is quite close to merging!

Thank you! I have fixed the issue where the reviews list page showed NULL ratings. I also added the reviews title to the the page. I also changed the ratings to be placed </5.00>. Will work on making the backlog issues.

**Comment by danielngira** [comment]
In my latest commit, I fixed a merge conflict in the `views.py` file

**Review by majorsylvie — APPROVED** [review]
You've successfully added the title, `5.00`, and removed the rating text for non-existent ratings! 

Great work! Lets get this merged.

**Comment by majorsylvie** [comment]
Issue Score: Excellent

Comments:
Excellent work @danielngira ! 
You've got a descriptive summary and closing statement, you got and acted on feedback in your PR very well, you provided video evidence showing off your work.

Incredible job! 

Keep moving forward :)

**Pull request merged**